src/components/Shell.tsx

```diff
@@ -52,13 +54,51 @@ async function fetchStudyConfig(configLocation: string, configKey: string) {
   return parseStudyConfig(config, configKey);
 }
 
-type Props = {
-  globalConfig: GlobalConfig;
-};
+function orderObjectToList(
+  order: OrderObject, 
+  pathsFromFirebase: {
+    path: string;
+    order: string[];
+  }[] | undefined,
+ path: string
+ ) : string[] {
```

Comment:
Implement this in 2 functions, 1 to be called and 1 that is called recursively

Reply:
done